app/helpers/patient_helper.rb

```diff
@@ -30,17 +30,9 @@ def normalize_and_get_state_name(name)
   end
 
   def time_zone_offset_for_state(name)
-    # Grab state time zone information by name
-    state = states_with_time_zone_data[normalize_name(name)]
-
-    # Grab time zone offset
-    offset = state.nil? ? -4 : state[:offset]
-
-    # Adjust for DST (if observed)
-    offset -= 1 if state && state[:observes_dst] && !Time.use_zone('Eastern Time (US & Canada)') { Time.now.dst? }
-
-    # Format and return the offset
-    (offset.negative? ? '' : '+') + format('%<offset>.2d', offset: offset) + ':00'
+    # Call TimeZone#now to create a TimeWithZone object that will contextualize
+    # the time to the current truth
+    ActiveSupport::TimeZone[time_zone_for_state(name)].now.formatted_offset
```

Comment:
@Bialogs Is there an official list on the time zone names that are supported by `ActiveSupport::TimeZone`? I'm wondering to make sure that no zones we use are left uncovered.

Reply:
It looks like if you supply a bad string to `ActiveSupport::TimeZone[]` like `ActiveSupport::TimeZone['foo']` then it returns nil. Could we add in a test to make sure that all of our timezones return non-nil when calling this?

Reply:
Yes you have to go back to the tzinfo gem to see the full list of zone names...

```ruby
require 'tzinfo'
TZInfo::Timezone.all
```

Reply:
Ok, thanks for the link. I think it's worth adding the test that loops through all timezones from https://github.com/SaraAlert/SaraAlert/blob/master/lib/assets/states_with_time_zone_data.yml in the case that it grows over time and we don't miss any missing timezones and/or typos that might exist